### PR TITLE
Test cases for rider app

### PR DESCRIPTION
**Pull request by Jaceken**
The initial version of test specs [rider app]

### CODE REVIEW

File: Test specification/Test cases/Milestone_1/Publisher app/1_General.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+### [General_001]
+---
+*  **Test case name:**  _Mapbox access token._
+*  Description: May embed the Mapbox access token in the app.
+*  **Expected result:** _The map is available and adequately presented to the user._
+
+### [General_002]
+---
+*  **Test case name:**  _Creating Trackable ID._
+*  Description: User can to create trackabel ID.
```
Comment (davyskiba) [inline Test specification/Test cases/Milestone_1/Publisher app/1_General.md]: Actually what we are doing is creating a trackable (soon to be renamed to order) with a user-entered id, which is labeled as `name` in the UI

Comment (davyskiba) [inline Test specification/Test cases/Milestone_1/Publisher app/1_General.md]:
```suggestion
*  Description: User can create trackable ID.
```

File: Test specification/Test cases/Milestone_1/Publisher app/2_Permissions.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+### [Permissions_004] [ANDROID]
+---
+*  **Test case name:**  _Revoking GPS permissions by Android._
+*  Description: Check if the user can give permissions when they are removed by Android's "Auto revoke permissions" option. [TIP] this mechanism is triggered when the app isn't used for a long time.
+*  **Expected result:** _Permissions question is presented to the user when the app is used next time._
+
+### [Permissions_005]
+---
+*  **Test case name:**  _Appkileld while GPS permission dialog is presented._
```
Comment (davyskiba) [inline Test specification/Test cases/Milestone_1/Publisher app/2_Permissions.md]:
```suggestion
*  **Test case name:**  _App killed while GPS permission dialog is presented._
```

File: Test specification/Test cases/Milestone_1/Publisher app/3_GPS.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+* **Expected result:** _Location is updated correctly._
+
+### [GPS_002]
+---
+*  **Test case name:**  _App working while the device is locked._
+* Description: _The application is working correctly while the device is locked._
+* **Expected result:** _Location is updated correctly._
+
+### [GPS_003]
+--
```
Comment (davyskiba) [inline Test specification/Test cases/Milestone_1/Publisher app/3_GPS.md]: This line is displaying differently 🤔 
```suggestion
---
```

**Comment by davyskiba** [comment]
Do you think it would make sense to list and link those files in the top-level README?

**Comment by Jaceken** [comment]
> Do you think it would make sense to list and link those files in the top-level README?

We will [create a separate file](https://github.com/ably/asset-tracking-demos/issues/2) with test process description/tips. I think only this file needs to be mentioned in top-level README.

File: Test specification/Test cases/Milestone_1/Publisher app/1_General.md (hunk 1 above)
Comment (Jaceken) [reply Test specification/Test cases/Milestone_1/Publisher app/1_General.md]: All clear.

Comment (Jaceken) [reply Test specification/Test cases/Milestone_1/Publisher app/1_General.md]: Resolving on behalf of @davyskiba

Comment (Jaceken) [reply Test specification/Test cases/Milestone_1/Publisher app/1_General.md]: Resolving on behalf of @davyskiba

File: Test specification/Test cases/Milestone_1/Publisher app/2_Permissions.md (hunk 2 above)
Comment (Jaceken) [reply Test specification/Test cases/Milestone_1/Publisher app/2_Permissions.md]: Resolving on behalf of @davyskiba

File: Test specification/Test cases/Milestone_1/Publisher app/3_GPS.md (hunk 3 above)
Comment (Jaceken) [reply Test specification/Test cases/Milestone_1/Publisher app/3_GPS.md]: Resolving on behalf of @davyskiba

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
We should avoid spaces in folder names, primarily because of how ugly they look in `github.com` links when shared (which these are likely to be, often).

The folder names are also perhaps a little over-complicated, with superfluous elements, and lacking conformance with the terminology we use elsewhere (i.e. [here](https://github.com/ably/asset-tracking-backend-demo#related-demos)). I would suggest:

- `testing/`
  - `rider.md`
  - `subscriber.md`

Using H2 (`##`) headings in each file for each section (e.g. 'General', 'Permissions', etc..).

Using H3 (`###`) headings under those for each test case.

I would add a bullet to each test indicating what milestone it's for. This means you can also link to the milestones in app requirements.

Also, I'm not sure why you've got your current H3 headings enclosed in square brackets. That's not necessary for any reason I know of. 😕

**Review by QuintinWillison — APPROVED** [review]
Thanks for making those structural changes, @Jaceken - I appreciate you doing that for me. 🙇 

I'm not going to attempt to review the test cases in detail, nor am I making an effort - right now - to try to identify missed tests. On a quick skim [read](https://github.com/ably/asset-tracking-demos/tree/QA-test-documentation/testing), it's looking good, thanks.

This is a great starting point for our test plans, crucially presenting open visibility for our team and other visitors as to what we consider important to test when manually validating these demo apps. 🚀

**Review by ikbalkaya — APPROVED** [review]
Looks good :) I will go through this and add any additional test cases and submit them on a separate PR

**Pull request merged**